### PR TITLE
Add support for cobbler >= 3.3.3

### PR DESCRIPTION
**Pull request by meaksh**
## What does this PR change?

This PR does the necessary adjustments to work with latest cobbler versions >= 3.3.3:

- Adjust custom "sync_post_tftpd_proxies.py" cobbler module to new API.
- Remove deprecated `next_server` cobbler setting in favor of `next_server_v4` and `next_server_v6`.
- Adjust names for available fence agents
- Adjust testsuite scenarios to new fence agents names
- extend the already implemented Cobbler steps to include checking for profiles
- add tests for Cobbler `buildiso`

**IMPORTANT NOTE: DO NOT MERGE THIS UNTIL COBBLER 3.3.3 IS PUSHED TO UYUNI AND HEAD**

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- No documentation needed: only internal and user invisible changes
- Documentation issue was created: [Link for SUSE Manager contributors](https://github.com/SUSE/spacewalk/issues/new?template=ISSUE_TEMPLATE_DOCUMENTATION.md&labels=documentation&projects=SUSE/spacewalk/31), [Link for community contributors](https://github.com/uyuni-project/uyuni-docs/issues/new).
- (OPTIONAL) [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pulls)

- [ ] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Tracks https://github.com/SUSE/spacewalk/issues/16538
Fixes https://github.com/SUSE/spacewalk/issues/17250

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"  
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by SchoolGuy — CHANGES_REQUESTED** [review]
The [configure-tftpsync.sh script](https://github.com/uyuni-project/uyuni/blob/0aaf348aa3cad372206cac2b3381f17264b82c43/tftpsync/susemanager-tftpsync/configure-tftpsync.sh) was not modified, thus a setup wouldn't work.

Ideally we use `cobbler-setting` for modifying the settings file but this is blocked due to https://github.com/cobbler/cobbler/issues/2992

**Review by SchoolGuy — CHANGES_REQUESTED** [review]
Okay this is looking much better now.

However the following is missing:

- `spacewalk/setup/bin/spacewalk-setup-cobbler`: `COBBLER_CONFIG_FILES` needs to be adjusted from `settings` to `settings.yaml`
- `spacewalk/setup/spacewalk-setup.spec` needs to have the Cobbler Version requirement increased to 3.3.2 or greater
- `spacewalk/setup/spacewalk-setup.spec` needs to have the [workaround](https://github.com/uyuni-project/uyuni/pull/4393) from @sbluhm removed

**Review by SchoolGuy — APPROVED** [review]
We need to see what the tests are saying but this PR seems lgtm for me now.

**Comment by sbluhm** [comment]
Do you want me to test the kickstart process on an Alma8 instance or is this expected to work? If yes, is this ready to test or should I wait?

**Comment by SchoolGuy** [comment]
@sbluhm Sadly this is still not in a working state. Also we are blocked until I have resolved https://github.com/cobbler/cobbler/issues/3016 both upstream and downstream.

**Review by srbarrios — APPROVED** [review]
Approving test code changes.

**Comment by sbluhm** [comment]
@SchoolGuy thanks for the info. Offer still holds. Just let me know if/when.

**Comment by nodeg** [comment]
@SchoolGuy, @meaksh: FYI: I have created my Cobbler buildiso tests against this branch and opened another PR for it: #5142. When you are ready with this PR, please merge my PR first into `master-fixes-for-new-cobbler` and then merge yours into `master`.

**Comment by SchoolGuy** [comment]
@meaksh I found out that `uyuni/utils/spacewalk-hostname-rename` utilises the old settings file. Thus this change is missing on this PR. I will prepare a commit for this later.

**Comment by SchoolGuy** [comment]
I rebased this onto `master` and pushed a few of my WIP changes along the way. Let's see how the testsuites like that while I continue implementing the new additional object features.

**Comment by SchoolGuy** [comment]
Rebased onto `master` to solve the conflicts.

**Review by nodeg — APPROVED** [review]
The code for testing Cobbler buildiso was written by me and tested in a local sumaform env. before merging it into this branch. So that part LGTM.

File: java/code/src/com/redhat/rhn/common/conf/ConfigDefaults.java
Comment (agraul) [inline java/code/src/com/redhat/rhn/common/conf/ConfigDefaults.java:539]: should this stay `getInt()` even though the method returns a `double`?

File: tftpsync/susemanager-tftpsync/sync_post_tftpd_proxies.py
Comment (agraul) [inline tftpsync/susemanager-tftpsync/sync_post_tftpd_proxies.py:79]: This could be logged lazily.

File: tftpsync/susemanager-tftpsync/sync_post_tftpd_proxies.py
Comment (agraul) [inline tftpsync/susemanager-tftpsync/sync_post_tftpd_proxies.py:105]: This could be logged lazily.

File: tftpsync/susemanager-tftpsync/sync_post_tftpd_proxies.py
Comment (agraul) [inline tftpsync/susemanager-tftpsync/sync_post_tftpd_proxies.py:184]: This could be logged lazily.

File: tftpsync/susemanager-tftpsync/sync_post_tftpd_proxies.py
Comment (agraul) [inline tftpsync/susemanager-tftpsync/sync_post_tftpd_proxies.py:233]: This could be logged lazily.

File: tftpsync/susemanager-tftpsync/sync_post_tftpd_proxies.py
Comment (agraul) [inline tftpsync/susemanager-tftpsync/sync_post_tftpd_proxies.py:237]: This could be logged lazily.

File: tftpsync/susemanager-tftpsync/sync_post_tftpd_proxies.py
Comment (agraul) [inline tftpsync/susemanager-tftpsync/sync_post_tftpd_proxies.py:246]: This could be logged lazily.

**Review by agraul — COMMENTED** [review]
I have one question about the Java code, but I only looked at it briefly. Other than that, I only have my common comment about lazy logging.

**Comment by SchoolGuy** [comment]
@sbluhm I am not 100% sure but I believe testing this PR is of interest for you.

**Comment by sbluhm** [comment]
YES! So I guess it is ready for testing? Unfortunately EL9 doesn't install anymore due to a PostgreSQL issue so might take some time.

**Comment by sbluhm** [comment]
@SchoolGuy ok that is probably a bit tight for me. So will test in my own time and submit changes to master as usual. Thanks for this one!

**Review by agraul — APPROVED** [review]
All my comments were addressed.

File: java/code/src/com/redhat/rhn/domain/kickstart/KickstartData.java
Comment (lucidd) [inline java/code/src/com/redhat/rhn/domain/kickstart/KickstartData.java:1583]: nitpick but this can be reduced to 
```java
       return prof.getKsMeta()
                .map(ksm -> RegistrationType.find((String)ksm.get(RegistrationType.COBBLER_VAR)))
                .orElseGet(RegistrationType::getDefault)
```
getting rid of the extra variable and .get() call.

**Review by lucidd — APPROVED** [review]

**Review by srbarrios — APPROVED** [review]
TestSuite side LGTM

**Review by aaannz — APPROVED** [review]

File: java/code/src/com/redhat/rhn/common/conf/Config.java
Comment (SchoolGuy) [reply java/code/src/com/redhat/rhn/common/conf/Config.java:348]: Yes. Primitive objects are not nullable and thus I can't return a primitive object above. Since here in all cases I have a primitive object I want to use that. This pattern is also used for `float` in the lines above already.

**Review by cbbayburt — APPROVED** [review]
LGTM 👍

**Review by rjmateus — APPROVED** [review]
I didn't give a dip look, but with high level view looks good to me.

**Comment by SchoolGuy** [comment]
@sbluhm Since some test failed we pushed this to 4.3.2, thus you know have a lot of time to test. I don't expect changes in the code atm, only fixes to the Java Unit-Tests. Feel free to prove me wrong. :D

**Comment by SchoolGuy** [comment]
@sbluhm Addendum: I prepared a project for you to test - https://build.opensuse.org/project/show/home:SchoolGuy:branches:systemsmanagement:Uyuni:Master

**Comment by SchoolGuy** [comment]
This seems stable enough now. Merging it now.

**Pull request merged**
